test/generateTasks.spec.js

```diff
@@ -176,3 +191,41 @@ describe('generateTasks', () => {
     })
   })
 })
+
+describe('generateTasks Special Cases', () => {
+  beforeAll(() => {
+    jest.spyOn(process, 'cwd').mockReturnValue(path.join(gitDir, 'sub-dir'))
+  })
+
+  afterAll(() => {
+    process.cwd.mockRestore()
+  })
+
+  it('should not match non-children files', async () => {
+    const relPath = path.join(process.cwd(), '..')
+    const result = await generateTasks({ config, gitDir: relPath, files: filesSpecialCases })
```

Comment:
`generateTasks` should accept the `cwd` as an argument, so is the mock value necessary? Might make the test more readable.